src/EdFi.AnalyticsMiddleTier.DataStandard32/Engage/MSSQL/0006-Upcoming-Submissions-Map.sql

```diff
@@ -0,0 +1,45 @@
+﻿-- SPDX-License-Identifier: Apache-2.0
+-- Licensed to the Ed-Fi Alliance under one or more agreements.
+-- The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+-- See the LICENSE and NOTICES files in the project root for more information.
+
+WITH UpcomingSubmission as (
+	SELECT 
+		DescriptorConstant.DescriptorConstantId,
+		d.DescriptorId
+	FROM 
+		analytics_config.DescriptorConstant
+	CROSS JOIN (
+		SELECT 
+			Descriptor.DescriptorId
+		FROM
+			edfi.Descriptor
+		INNER JOIN
+			lmsx.AssignmentCategoryDescriptor ON
+				Descriptor.DescriptorId = AssignmentCategoryDescriptor.AssignmentCategoryDescriptorId
+		WHERE
+			Descriptor.CodeValue = 'UpcomingSubmission'
```

Comment:
Google actually has [three submission states](https://developers.google.com/classroom/reference/rest/v1/courses.courseWork.studentSubmissions#SubmissionState) for this, which we ought to be using in the extractor: "new", "created", and "reclaimed_by_student".

Right now the Google extractor is automatically changing those all to "missing", which doesn't seem right. It should probably only be "missing" if the course work is also _late_.

So in the extractor, I think we need to simplify a little bit. We're hiding too much of the original information provided by Google. We probably still need to convert those three values to "missing" if the assignment is late, but otherwise either keep those three values or change to "unsubmitted" or "upcoming" or something. My preference is to modify as little as possible. In that case, the Extractor should return to sending out those three states, unmodified.

But then what about the Harmonizer, when it is matching up to descriptors? Well, in fact we already have those three values as descriptors for Google! Therefore the Descriptor mapping here needs to look for descriptor code values of `('UpcomingSubmission', 'CREATED', 'NEW', 'RECLAIMED_BY_STUDENT')`.

We do not have a similar descriptor for Canvas, so it needs to have a new value.

After all this analysis, I think the term "UpcomingSubmission" needs to be a little simpler and shorter. Other options include:

* New
* Upcoming
* Future
* Pending
* Assigned

I'm leaning a little toward "Assigned." 

@bradbanister would like to get your views as well as @lalrg's on this diatribe.

Reply:
I changed the constant name to Upcoming, sounds good to me... And also updated the mappings